src/metabase/models/serialization.clj

```diff
@@ -91,8 +91,10 @@
   done if a need arises."
   [model-name eid]
   (let [model (keyword "model" model-name)
-        pk    (first (t2/primary-keys model))]
-    (t2/select-one-fn pk [model pk] :entity_id eid)) )
+        pk    (first (t2/primary-keys model))
+        eid   (cond-> eid
+                (str/starts-with? eid "eid:") (subs 4))]
```

Comment:
A bit sucky that we coerce numbers to strings here, I'd first do a type check.

Reply:
eh, which numbers? don't understand the comment.

Reply:
Would an id be a long here? If so this is implicitly converting it to a string and then throwing that string away, just wasteful work.

Reply:
Brain fart on my side, as its not overloaded here.